CMakeLists.txt

```diff
@@ -25,21 +25,40 @@ remove_definitions(-DQT_NO_CAST_FROM_ASCII)
 
 find_package(Qt5 CONFIG REQUIRED
     Core
-    Network
-    Widgets
+    Gui
     Sql
+    Widgets
+    Xml
 )
 
 find_package(KF5 REQUIRED COMPONENTS
-    Contacts
+    Config
+    ConfigWidgets
+    CoreAddons
     I18n
+    WidgetsAddons
```

Comment:
Why do I need the Config, ConfigWidgets, CoreAddons and WidgetsAddons? Are these dependencies of ...XmlGui?

Reply:
You're using various headers from frameworks that are brought in implicitly by kdelibs4support. Once you remove kdelibs4support, you will also have to properly link against those.

* Config: `KConfigGroup` etc.
* ConfigWidgets: `klanguagebutton.h` etc.
... and so on :)

Reply:
Hmm, ok, that is not intended. All kdelibs4support needs to go.

Do you know a reliable method of finding all these wrong headers? Could you pass me a list?

Reply:
Well, kdelibs4support is not even a release blocker for KDE Plasma or Applications. Nevertheless good to get rid of. Please note that the headers likely are not (all) wrong. Config, ConfigWidgets etc. are proper Frameworks and I only assumed you were able to use them because you somewhere brought in kdelibs4support (which just adds a lot of the frameworks to its public link interface). I've made the patch a while ago... and what probably happened was me trying to build kraft-kf5 on a system without any kdelibs4 and qt4 present, and running into build failures.

Reply:
Since we build with `--as-needed`, the list of Frameworks I've added to your cmake file represents the real-world linkage of Frameworks in your application.

Reply:
Ok, I pushed a few more fixes to branch `kf5`. Could you rebase this PR and check if you still need `Config`, `ConfigWidgets`? 

`CoreAddons` will still be needed for at least KAboutData, which I haven't removed yet. With `WidgetsAddons` I am not sure, I removed a few bad headers, but not sure if that is enough.

Thanks for your help!

Reply:
It seems that KWidgetsAddons is a dependency of akonadi contact module.